biz.aQute.junit/src/aQute/junit/ConsoleReporter.java

```diff
@@ -0,0 +1,57 @@
+package aQute.junit;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import aQute.lib.exceptions.Exceptions;
+import junit.framework.AssertionFailedError;
+import junit.framework.Test;
+import junit.framework.TestListener;
+
+public class ConsoleReporter implements TestListener {
+
+	private class State {
+		Throwable				t;
+		AssertionFailedError	a;
+	}
+
+	private final ConcurrentMap<Test,State> testMap = new ConcurrentHashMap<Test,ConsoleReporter.State>();
+
+	@Override
+	public void addError(Test test, Throwable t) {
+		testMap.putIfAbsent(test, new State()).t = t;
+	}
+
+	@Override
+	public void addFailure(Test test, AssertionFailedError a) {
+		testMap.putIfAbsent(test, new State()).a = a;
+	}
+
+	@Override
+	public void endTest(Test test) {
+		StringBuilder sb = new StringBuilder();
```

Comment:
What is the value of using a StringBuilder in this method when you just print it out at the end of the method? Why not just System.out.print as you go instead of sb.append?

In fact, you could do it all with a single print if you add a toString method to State which handles its string value.